src/tests/add.rs

```diff
@@ -129,3 +132,103 @@ fn add_key_to_session() {
     assert_eq!(new_payload, new_expected);
     keyring.unlink_key(&key).unwrap();
 }
+
+#[test]
+fn add_encrypted_key_to_session() {
```

Comment:
I'd rather that keytype-specific tests go into `src/keytypes/TYPE.rs` than at the operation level.